include/index.h

```diff
@@ -283,6 +287,9 @@ template <typename T, typename TagT = uint32_t, typename LabelT = uint32_t> clas
     // Acquire exclusive _tag_lock and _delete_lock before calling
     int reserve_location();
 
+    // reposition medoids/frozen points for dynamic filtered index.
+    void reposition_medoids(uint32_t old_location_start, uint32_t new_location_start);
```

Comment:
wasn't there such a function even for unfiltered dynamic index?

Reply:
Yes, there is reposition_frozen_points(), but in this case actually we are nor repositioning anything we are just updating respective medoid id with already repositioned frozen points, so that they would align. We can rename the function as update_medoid_location or something like that.